citations-tool/tool/src/webapp/js/citationscript.js

```diff
@@ -671,11 +671,34 @@ function checkRequiredFields( alertMsg ) {
     return false;
   }
   
+  //If electronic citation check if at least one url is supplied
+  if($('#type_selector').val() == 'electronic'){
+      var showUrlAlert = $("input:text[name^='url_']" ).filter(function(){
+                         return this.value.trim().length
+                         }).length > 0;
+      if(!showUrlAlert){
+        alert( "Must supply at least one url for electronic citation." );
+        return false;
+      }
+  }
   return true;
 }
 
+function checkReqStarForElectronicCitation(){
+  //For electronic citation display reqstar for Link field.
+  if($('#type_selector').val() == 'electronic'){
+      var addReqStar = '<span class="reqStar">*</span>\n';
+      $(addReqStar).insertBefore('#url_div label:first');
+  }
+  //else remove reqStar from the Link field
+  else{
+      $('#url_div .reqStar').remove();
+  }
+}
+
 $(document).ready( function() {
     if (typeof $.fn.googleBooksCover != "undefined"){
         $.fn.googleBooksCover();
     }
+    checkReqStarForElectronicCitation();
```

Comment:
$document.ready isn't called when you change the dropdown from e.g., book to electronic citation